src/instructlab/lab.py

```diff
@@ -712,15 +712,36 @@ def chat(
 def download(ctx, repository, release, filename, model_dir):
     """Download the model(s) to train"""
     click.echo(f"Downloading model from {repository}@{release} to {model_dir}...")
+    if (
+        "HF_TOKEN" not in os.environ
+        and repository != "instructlab/merlinite-7b-lab-GGUF"
```

Comment:
do we need this check? I think it might be best not to hardcode merlinite for such checks - it would also be good practice to always have an HF token exposed and would make the code easier and cleaner to maintain (even if technically you dont need a token for the instructlab org)

just me 0.02

Reply:
I assume this was so at least the default config didn't require it, which is nice.

An even better UX would determine whether the token is required for a given model before then erroring out because it's not set, but that could be a future improvement ...

Reply:
right - then for now should we at least only check if "instructlab" is part of the repo string, and we allow the download without token if true? since we would want to at least have uniform behavior for all the models in the `instructlab` HF org

Reply:
How about you use clicks' feature to read [values from env vars](https://click.palletsprojects.com/en/8.1.x/options/#values-from-environment-variables)?
`@click.option('--hf-token', envvar='HF_TOKEN')`

Reply:
merl is the default. This is my way of checking `If NO HF TOKEN and THE USER DIDN'T SET --FILENAME`

Reply:
@tiran yes! I was just looking at that yesterday when looking at https://github.com/instructlab/instructlab/pull/1022

Reply:
I was surprised by it asking me for a HF_TOKEN when I did `ilab download --repository instructlab/merlinite-7b-lab` as that repository does not need one. In my case, I specifically wanted it to download the non-gguf version of this model, but perhaps the logic around HF_TOKEN usage should be a bit broader?

Reply:
This was improved further in https://github.com/instructlab/instructlab/pull/1035